codec/decode.go

```diff
@@ -3,86 +3,101 @@ package codec
 import (
 	"encoding/binary"
 	"errors"
+	"reflect"
 )
 
-// DecodeInteger accepts a byte array representing a SCALE encoded integer and performs SCALE decoding of the
-// int, then returns it
-func DecodeInteger(b []byte) (int64, error) {
-	if len(b) == 1 {
-		return int64(b[0] >> 2), nil
-	} else if len(b) == 2 {
+// DecodeInteger accepts a byte array representing a SCALE encoded integer and performs SCALE decoding of the int
+// if the encoding is valid, it then returns (o, bytesDecoded, err) where o is the decoded integer, bytesDecoded is the
+// number of input bytes decoded, and err is nil
```

Comment:
still confused about what bytesDecoded is. number of bytes is the length of the decoded integer?

Reply:
bytesDecoded is the total number of bytes that were used to create the decoded integer so this includes the mode/length byte